### PR TITLE
Create flutter-ci.yml

### LINKED ISSUE #11: Add GitHub action to run dartfmt upon merge with master
**Please describe the feature you want this project to offer**
It'd be nice to have consistent and formatted code in the master branch to enhance code readability and navigation.

**Describe the solution you'd like**
Make a GitHub action which would run when a PR gets merged to master. The action would run `dartfmt` to format the code according to the official coding style guideliness.

### PR DESCRIPTION
**Pull request by Mastersam07**
# Story Title

[Add GitHub action to format code upon merge with master](https://github.com/IEEE-VIT/hacktoberfest-flutter/issues/11)

## Changes made

-   added flutter-ci.yaml
-   run flutter analyze . and format code on pull request closed and merged

## How does the solution address the problem

This PR will check for closed pull requests that were merged to master, run flutter linting analysis and format codebase.

## Linked issues

Resolves #11

### CODE REVIEW

File: .github/workflows/flutter-ci.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+name: Flutter CI
+
+on:
+  pull_request:
+    branches:
+      - master
+    types: [closed]
+
```
Comment (DarthBenro008) [inline .github/workflows/flutter-ci.yml:8]: kindly change this to pull request of any kind to master.

File: .github/workflows/flutter-ci.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+env:
+  flutter_version: "1.20.4"
+  java_version: "12.x"
+
+jobs:
+  format-and-lint:
+    if: github.event.pull_request.merged == true
```
Comment (DarthBenro008) [inline .github/workflows/flutter-ci.yml:15]: please remove this check

**Review by DarthBenro008 — CHANGES_REQUESTED** [review]
@Mastersam07 so there was a internal discussion and its been final that the CI should check lint on every PR and Push to master branch, hence make the changes accordingly, everything else seems good. Nice Work! :)

**Comment by Mastersam07** [comment]
@DarthBenro008 how about now?

**Review by DarthBenro008 — APPROVED** [review]
LGTM!

**Pull request merged**